### ISSUE TITLE
FlxSpriteUtil.space param spaceFromBounds is doing nothing

### ISSUE THREAD

**Issue by Tembac** [HaxeFlixel/flixel#1963]
- Flixel version: 4.1.1
- OpenFL version: 3.6.1
- Lime version: 2.9.1
- Affected targets: all

---

**Code snippet reproducing the issue**:

``` haxe

var spr1:FlxSprite = new FlxSprite();
spr1.makeGraphic(20, 20);
add(spr1);

var spr2:FlxSprite = new FlxSprite();
spr2.makeGraphic(20, 20);
add(spr2);

var spr3:FlxSprite = new FlxSprite();
spr3.makeGraphic(20, 20);
add(spr3);

FlxSpriteUtil.space([spr1, spr2, spr3], 0, 0, 10, 0, true);

```

---

**Observed behavior:**
![](http://puu.sh/rquyZ/82d6aad396.png)

**Expected behavior:**
![](http://puu.sh/rquCD/71fa1a75fb.png)
##

**Comment by Gama11** [comment]
Interesting... `prevWidth` and `prevHeight` in that function are never set. The original Flixel Power Tools had the same issue though, this is just a straight port from that:

https://github.com/photonstorm/Flixel-Power-Tools/blob/0a37c8e7a3c6133adef1ab090cac5e60e3939133/src/org/flixel/plugin/photonstorm/FlxDisplay.as#L135

**Comment by Gama11** [comment]
From the description of `spaceFromBounds`, I'm actually not 100% sure how this was supposed to work. Presumably it should look like this?

```diff
        public static function space(objects:Array<FlxObject>, startX:Int, startY:Int, horizontalSpacing:Int = 0,
                verticalSpacing:Int = 0, spaceFromBounds:Bool = false):Void
        {
-               var prevWidth:Int = 0;
-               var prevHeight:Int = 0;
+               var prevWidth:Float = 0;
+               var prevHeight:Float = 0;

                for (i in 0...objects.length)
                {
                        var object = objects[i];

                        if (spaceFromBounds)
                        {
                                object.x = startX + prevWidth + (i * horizontalSpacing);
                                object.y = startY + prevHeight + (i * verticalSpacing);
+
+                               prevWidth = object.width;
+                               prevHeight = object.height;
                        }
                        else
                        {
                                object.x = startX + (i * horizontalSpacing);
                                object.y = startY + (i * verticalSpacing);
                        }
                }
        }
```

However, the result of that looks like this:

![](http://i.imgur.com/LuuEeP1.png)

**Comment by Gama11** [comment]
What makes you think the expected behavior should look like in your screenshot?

**Comment by Tembac** [comment]
Because that's what I need 😆 I don't know if it is what was intended.

I was looking for something similar to this on Phaser: http://phaser.io/examples/v2/groups/align-sprites-to-grid

http://phaser.io/docs/2.6.2/Phaser.Group.html#align

**Comment by SamBumgardner** [comment]
I'm not sure if anyone is still watching this issue, but I think I can see what's going on.

You want the sprites to have 10 pixels space between them, but the offset specified in `FlxSpriteUtil.space([spr1, spr2, spr3], 0, 0, 10, 0, true);` just separates each sprite's *origin* by 10 pixels. You end up with the origin of each sprite only being 10 pixels apart and the graphic is 20 pixels wide, which makes them overlap each other.

 To get the amount of space you want, make the offset be your sprite's width + the amount of space you want between them, like this:

`FlxSpriteUtil.space([spr1, spr2, spr3], 0, 0, spr1.width + 10, 0, true);`

If you use the above code, keep in mind that if you change the size of `spr1`'s hitbox, the spacing between the sprites will change as well. It may be worth just keeping a variable somewhere that stores the width/height of your tiles instead.

**Comment by Gama11** [comment]
I think this should be resolved by @Skerper's changes in #2154.
